agent/internal/binding-accessor/binding-accessor.go

```diff
@@ -33,10 +34,19 @@ type BindingAccessorFunc func(ctx Context) (value interface{}, err error)
 // be combined together in order to represent a whole binding accessor
 // expression through different function closures (field access, array access,
 // map access, etc.).
-type valueFunc func(ctx Context) (value interface{})
+type valueFunc func(ctx Context, maxDepth int) (value interface{})
 
-// A transformationFunc value transforms an input value to another output value
-type transformationFunc func(valueIn interface{}) (valueOut interface{})
+// A transformationFunc value transforms an input value to another output value.
+// The traversal of the input value cannot be deeper than maxDepth, and the
+// output value cannot exceed maxElements.
+type transformationFunc func(ctx Context, valueIn interface{}, maxDepth, maxElements int) (valueOut interface{})
+
+// Maximum binding accessor execution depth. The binding accessor execution
+// traverses Go values. It cannot go deeper than this value. It panics a
```

Comment:
It panics? Shouldn't it bubble up an error instead?

Reply:
fixed in 9dd8236f1100ed579f8824b2468306d21d5e2608